include/oxen/quic/udp.hpp

```diff
@@ -35,23 +35,25 @@ namespace oxen::quic
     {
         Path path;
         ngtcp2_pkt_info pkt_info{};
-        std::variant<bstring_view, bstring> pkt_data;
+        std::vector<std::byte> pkt_data;
```

Comment:
I'm concerned about how this change impacts manual packet routing: currently that code uses the bstring_view constructor, which means we can process the packet without needing to copy it first.  With this change, however, the packet data will have to get alloced + copied for every packet, which seems virtually certain to make manual packet routing slower.

I'm not entirely across what the intention of this code was; if it was just to eliminate use of bstring, then this should become a `std::variant<std::vector<std::byte>, std::span<const std::byte>>` instead to still allowing avoid that copy.

Reply:
(That said: this is my best guess as to what in this PR might cause the increased CPU usage; I didn't try isolating *just* this change from the rest of the 0rtt PR)

Reply:
Just looking at the numbers again today, and apparently I didn't look closely enough last night: it's primarily the server user time where there's a fairly large difference (about 20% more CPU time), which supports this being a slowdown in packet processing (since the server does way more packet processing in a speedtest run).

Reply:
An alternative approach is, instead of a variant, have *both* vector and span members: when we need to copy, we copy into the vector and set the span to the vector; when we don't need a copy we just don't set the vector at all and set to the span to whatever input we were fed.  And then `data` can just directly use the span rather than needing to visit the two possibilities.

Reply:
You're exactly right, and thank you for doing the leg work on getting those results. We spoke about this off band; I was aiming to merge the oxenc PR and do the vector/span member. Will fix 👍